api/src/v2/Registration/Uploader/OrgDeputyshipUploader.php

```diff
@@ -303,11 +299,21 @@ private function handleReport(OrgDeputyshipDto $dto)
 
```

Comment:
I feel like this function has got too complex to read. Too many nested `if`'s statements. I _think_ we can simplify if we write something like this.

I haven't tested the below but if the function is the exact same as before I think going this way will help with readability

```
// rather than an else for if a report exists we can immediately 
// check if no report exists do what we have to do and then presist/flush
// and return out the function so it stops executing
if (!$report) {
    $report = new Report(
        $this->client,
        $dto->getReportType(),
        $dto->getReportStartDate(),
        $dto->getReportEndDate()
    );

    $this->client->addReport($report);

    $this->added['reports'][] = $this->client->getCaseNumber().'-'.$dto->getReportEndDate()->format('Y-m-d');

    $this->em->persist($report);
    $this->em->flush();

    return;
}

// next we checked if report was NOT submitted and if the un-submitted date was empty
// instead we check if the report has been submitted and that the un-submitted date is NOT empty
// so basically the opposite of the original conditions. If they are met we just return again.
// As we didn't have an else attached to this and we don't want to do anything if these are true.
if ($report->getSubmitted() && !empty($report->getUnSubmitDate())) {
    return;
}

// Now the rest of the function is the same as before just with 
// else nested if's and easier to follow and understand.
if (OrgDeputyshipDto::DUAL_TYPE == $dto->getHybrid()) {
    if ($this->client->getNamedDeputy()->getDeputyUid() == $dto->getDeputyUid()) {
        if ($report->getType() !== $dto->getReportType()) {
            $report->setType($dto->getReportType());

            $this->updated['reports'][] = $report->getId();
        }
    }
} else {
    if ($report->getType() !== $dto->getReportType()) {
        // Add audit logging for report type changing
        $report->setType($dto->getReportType());

        $this->updated['reports'][] = $report->getId();
    }
}

$this->em->persist($report);
$this->em->flush();
```